synapse/handlers/oidc.py

```diff
@@ -220,11 +220,11 @@ async def handle_oidc_callback(self, request: SynapseRequest) -> None:
                 session, state
             )
         except (MacaroonInitException, MacaroonDeserializationException, KeyError) as e:
-            logger.exception("Invalid session for OIDC callback")
+            logger.warning("Invalid session for OIDC callback: %s", e)
             self._sso_handler.render_error(request, "invalid_session", str(e))
             return
         except MacaroonInvalidSignatureException as e:
-            logger.exception("Could not verify session for OIDC callback")
```

Comment:
This means the signature on the macaroon was wrong, so either the server just rotated their macaroon secret, or they have something really wrong in their setup (like different workers not sharing the same secret). I would probably keep this one?

Reply:
It's still kept as a warning. Is the traceback useful to a server operator?

```
MacaroonInvalidSignatureException: Signatures do not match
  File "synapse/handlers/oidc.py", line 220, in handle_oidc_callback
    session, state
  File "synapse/handlers/oidc.py", line 1198, in verify_oidc_session_token
    v.verify(macaroon, self._macaroon_secret_key)
  File "pymacaroons/verifier.py", line 55, in verify
    discharge_macaroons
  File "pymacaroons/verifier.py", line 77, in verify_discharge
    raise MacaroonInvalidSignatureException('Signatures do not match')
```

With the proposed change, we would log the message `Invalid session for OIDC callback: Signatures do not match` as a warning.

I could throw in the name of e's class (i.e. `Invalid session for OIDC callback - MacaroonInvalidSingatureException: Signatures do not match`) if that's clearer?

Reply:
> With the proposed change, we would log the message Invalid session for OIDC callback: Signatures do not match as a warning.

That looks fine to me 👍